rerun_cpp/src/rerun/collection_adapter_builtins.hpp

```diff
@@ -21,75 +26,89 @@ namespace rerun {
         }
     };
 
-    /// Adapter from std::vector<T> where T can be converted to TElement
-    template <typename TElement, typename T>
+    /// Adapter for a iterable container (see `rerun::traits::is_iterable_v`) which
+    /// has a value type from which `TElement` can be constructed but is not equal to `TElement`.
+    ///
+    /// Since this needs to do a conversion, this will always need to allocate space.
+    /// However, if a temporary is passed, elements will be moved instead of copied upon construction of `TElement`.
+    template <typename TElement, typename TContainer>
     struct CollectionAdapter<
-        TElement, std::vector<T>,
+        TElement, TContainer,
         std::enable_if_t<
-            !std::is_same_v<TElement, T> && std::is_constructible_v<TElement, const T&>>> {
-        Collection<TElement> operator()(const std::vector<T>& input) {
-            std::vector<TElement> transformed(input.size());
-
-            std::transform(input.begin(), input.end(), transformed.begin(), [](const T& datum) {
-                return TElement(datum);
-            });
-
-            return Collection<TElement>::take_ownership(std::move(transformed));
+            !std::is_same_v<TElement, traits::value_type_of_t<TContainer>> && //
+            traits::is_iterable_v<TContainer> &&                              //
+            std::is_constructible_v<
+                TElement,
+                traits::value_type_of_t<TContainer>> //
+            >> {
+        Collection<TElement> operator()(const TContainer& input) {
+            std::vector<TElement> elements;
+            // Insert supposedly automatically reserves if size is known, see https://stackoverflow.com/a/35359472
+            elements.insert(elements.end(), std::begin(input), std::end(input));
```

Comment:
I'm pretty sure this also works

```suggestion
            std::vector<TElement> elements(std::begin(input), std::end(input));
```

Reply:
much nicer, yes!